File: solutions/javascript/pig-latin/1/pig-latin.js
//
// This is only a SKELETON file for the 'Pig Latin' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const translate = (sentence) => {
  const res = [];
  for(const word of sentence.split(' ')) {
    if(['a', 'e', 'o', 'i', 'u', 'xr', 'yt'].some(e => word.startsWith(e))) {
      res.push(word + 'ay');
      continue;
    }
    let i = 1;
    for(; i < word.length; i++) {
      if(word[i] === 'u' && word[i-1] === 'q')
        break;
      if(['a', 'e', 'o', 'i', 'u', 'y'].includes(word[i])) {
        i--;
        break;
      }
    }
    res.push(word.slice(i + 1) + word.slice(0, i + 1) + 'ay');
  }
  return res.join(' ');
};

File: solutions/python/pig-latin/1/pig_latin.py
def translate(text):
    translation = []
    for word in text.split(" "):
        if word.startswith(("a", "e", "i", "o", "u", "xr", "yt")):
            translation.append(word + "ay")
        else:
            end = ""
            for index, char in enumerate(word):
                if char == "u" and end.endswith("q"):
                    index += 1
                    end += char
                    break
                elif char in "aeiou":
                    break
                elif char == "y" and index != 0:
                    break
                end += char
            translation.append(word[index:] + end + "ay")
    return " ".join(translation)
            
        
